app/actions/local/group.ts

```diff
@@ -92,3 +94,37 @@ export const storeGroups = async (serverUrl: string, groups: Group[]) => {
     }
 };
 
+/**
+  * Store fetched groupTeams locally
+  *
+  * @param serverUrl string - The Server URL
+  * @param groups Group[] - The groups fetched from the API
+  * @param userId string - The member to associate groups with
+  * @param prepareOnly boolean - Prepare only without saving
+  */
+export const storeGroupMembershipsForMember = async (serverUrl: string, groups: Group[], userId: string, prepareOnly = false) => {
+    try {
+        const operator = getOperator(serverUrl);
+        const preparedModels: Array<GroupModel | GroupMembershipModel> = [];
+        const preparedGroups = await prepareGroups(operator, groups);
+
+        if (preparedGroups.length) {
+            preparedModels.push(...preparedGroups);
+        }
+
+        const {existingGroupMemberships, groupMemberships} = await prepareGroupMembershipsForUser(operator, groups, userId);
+        if (groupMemberships.length) {
```

Comment:
What if only `existingGroupMemeberships` has elements? Do we want to keep them?

Reply:
Yup discussed with @enahum last week - will change to only update existing ones, and for now - make the assumption that if no groupMemberships are returned, we delete existing ones.

Further PR's will also have various places where these will get updated / kept in sync.